### PR TITLE
Firebase subsciption mirror

### PR DESCRIPTION
**Pull request by dmdashenkov**
In this PR we transplant the Firebase subscription mirror from `todo-list` repository to this repository.

Also, in this PR we set up the publishing and code coverage scripts.

### CODE REVIEW

**Comment by dmdashenkov** [comment]
@armiol, PTAL.

File: .travis.yml [hunk 1]
```diff
@@ -2,6 +2,10 @@ language: java
 
 before_install:
   - chmod +x gradlew
+  - openssl aes-256-cbc -K $encrypted_7dcf2dcd412f_key -iv $encrypted_7dcf2dcd412f_iv -in secrets.tar.enc -out secrets.tar -d
+  - tar xvf secrets.tar
+  - mkdir ./firebase-mirror/src/test/resources
```
Comment (armiol) [inline .travis.yml]: Let's comment. These `mkdir` and `mv` need that.

File: firebase-mirror/src/main/java/io/spine/server/firebase/DocumentKeys.java [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+     * Transforms the given string to a <a target="_blank"
+     * href="https://firebase.google.com/docs/firestore/quotas">valid^</a> Firestore key.
+     *
+     * <p>A valid key should contain only English letters, digits and underscore symbols and NOT
+     * start with an underscore.
+     *
+     * <p>All the illegal chars are deleted from the source string. If no characters are left after
+     * this operation, an {@link IllegalArgumentException} is thrown.
+     *
+     * @param dirtyKey the string to transform
```
Comment (armiol) [inline firebase-mirror/src/main/java/io/spine/server/firebase/DocumentKeys.java]: `originalKey` would be better.

**Review by armiol — APPROVED** [review]
@dmdashenkov LGMT with some minor exceptions.

Please address the issues and feel free to merge.

**Pull request merged**